### DISCUSSION TITLE
Multiexpression Outputs Question

### DISCUSSION THREAD

**Discussion by ccclalala123** [MilesCranmer/PySR#705] [Q&A] [answered]
When I define multiexpression outputs, how do I know which is P and which is Q?
```
objective = """
function my_loss_function(tree, dataset::Dataset{T,L}, options)::L where {T,L}

  tree.degree != 2 && return L(Inf)

    P = tree.l
    Q = tree.r

    # Evaluate numerator:
    P_prediction, flag = eval_tree_array(P, dataset.X, options)
    !flag && return L(Inf)

    # Evaluate denominator:
    Q_prediction, flag = eval_tree_array(Q, dataset.X, options)
    !flag && return L(Inf)

    R12_2D =  (P_prediction .- 1)  .* cz1_jl ./ (0.09 .* woo_tran_jl .* (Q_prediction .+ c_pe_tran_jl .- 1.0))
    
    diffs = R12_2D .- dataset.y
    loss = sum(diffs .^ 2) 

    return loss
 
end
"""

model = PySRRegressor(
    population_size=200,
    niterations=200,
    binary_operators=["+", "-", "*", "/"],
    loss_function=objective,
)
model.fit(X, y)
model.get_best().equation
```
The results are as follows：
```
Complexity,Loss,Equation
1,Inf,"x0"
3,1.8108e-6,"0.6919551 + 1.4003153"
5,1.6049491e-6,"(0.43112686 - x0) * 2.6639194"
7,1.3138252e-6,"(x0 / (-0.3637128 - x0)) * 4.5015717"
9,1.048514e-6,"((x0 / (-0.32570758 - x0)) + x0) / 4.502915"
11,1.0421031e-6,"(((x0 / (-0.32833982 - x0)) + x0) / 0.90415287) - 4.490343"
```

**Comment by MilesCranmer** [comment, marked as answer]
Basically .l is the left side and .r is the right side of the outermost operator. So if you were numerically evaluating the equation, the last operator that you would evaluate is the one you split at. So `x0 / (-0.3637128 - x0)) * 4.5015717` has P equal to the x0/(-0.36 - x0) and Q equal to the 4.5. The outermost operator `*` is ignored.

But yes I think it would be nice for PySR to have a proper way of facilitating this, rather than needing to do it all manually.

**Reply by ccclalala123** [reply]
I got it! Thank you so much for your quick response!

**Reply by ccclalala123** [reply]
If there are 3 or more multi-expression outputs, how to realize and distinguish them? For example, Y=P×2+Q×3+W×4+T×5+K×6，where P, Q, W, T and K are function expressions of x.

**Reply by MilesCranmer** [reply]
Want to make a new thread for that? But basically just apply the pattern recursively

**Reply by ccclalala123** [reply]
How to achieve it in a simple way? Because I need to train 12 expressions of coefficients in an equation.

**Reply by Alexia369** [reply]
> How to achieve it in a simple way? Because I need to train 12 expressions of coefficients in an equation.

Do you have the code to achieve it？ I have met the same question
